src/azure-account.ts

```diff
@@ -40,16 +41,48 @@ function getNodeModule<T>(moduleName: string): T | undefined {
 	return undefined;
 }
 
+//List of environment names for user to choose from
+function getEnvironmentList(): string[] {
+	var list : string[];
+	list = ["Azure", "Azure US Government", "Azure China", "Azure Germany"];
+	return list;
+}
+
+//Returns AzureEnvironment that use has chosen
+function getEnvironment(name : string): any | undefined {
+	var environmentList : { [selectedname: string]: any} = {};
+	environmentList["Azure"] = (<any>AzureEnvironment).Azure;
+	environmentList["Azure US Government"] = (<any>AzureEnvironment).AzureUSGovernment;
+	environmentList["Azure China"] = (<any>AzureEnvironment).AzureChina;
+	environmentList["Azure Germany"] = (<any>AzureEnvironment).AzureGermanCloud;
+	if (name) {
+		return environmentList[name];
+	} else {
+		return environmentList["Azure"];
+	}
+}
+
+//placeholder for selectedEnvironment
+selectedEnvironment = (<any>AzureEnvironment).Azure;
+
+//check if setenvironment exists in keytar
+if(keytar.findPassword("storedEnvironment") == null){
+	keytar.setPassword("storedEnvironment", "current", "Azure");
+}
+
+//set selectedenvironment to current saved in keytar
+var storedenv = keytar.getPassword("storedEnvironment", "current");
+storedenv.then(function(env) {
+    selectedEnvironment = getEnvironment(env);
+});
+vscode.window.setStatusBarMessage("Current Azure Environment: " + selectedEnvironment.name);
+
 const logVerbose = false;
-const defaultEnvironment: AzureEnvironment = (<any>AzureEnvironment).Azure;
 const commonTenantId = 'common';
-const authorityHostUrl = defaultEnvironment.activeDirectoryEndpointUrl; // Testing: 'https://login.windows-ppe.net/'
 const clientId = 'aebc6443-996d-45c2-90f0-388ff96faa56'; // VSC: 'aebc6443-996d-45c2-90f0-388ff96faa56'
 const validateAuthority = true;
-const authorityUrl = `${authorityHostUrl}${commonTenantId}`;
-const resource = defaultEnvironment.activeDirectoryResourceId;
 
-const credentialsService = 'VSCode Public Azure';
+const credentialsService = selectedEnvironment.name;
```

Comment:
Lets keep that 'VSCode Public Azure'. It might have been a mistake on my part to have 'Public' in the name there. This way we only need to manage one refresh token.